File: INFS4102_project-main/Backend/eshop/src/main/java/qa/udst/eshop/models/NutritionProduct.java
package qa.udst.eshop.models;

import jakarta.validation.constraints.*;

//Nutrition Model: + Weight(Kg) minimal 0.03
public class NutritionProduct extends Product{
    @NotNull
    @DecimalMin("0.03")
    private double weightKg;

    @DecimalMin("0.0")
    private double calories;


    public NutritionProduct(String name, double price, ProductCategory category, String desc,
            @NotNull @DecimalMin("0.03") double weightKg, @NotBlank String imageURL, @DecimalMin("0.0") double calories) {
        super(name, price, category, desc, imageURL);
        this.weightKg = weightKg;
        this.calories = calories;
    }

    public double getWeightKg() {
        return weightKg;
    }

    public void setWeightKg(double weightKg) {
        this.weightKg = weightKg;
    }
    
    public double getCalories() {
        return calories;
    }

    public void setCalories(double calories) {
        this.calories = calories;
    }

}

File: models/NutritionProduct.dart
import 'Product.dart';

class NutritionProduct extends Product {
  final double weightKg;
  final double calories;

  const NutritionProduct({
    required super.id,
    required super.name,
    required super.price,
    required super.category,
    required super.desc,
    required super.imageURL,
    required this.weightKg,
    required this.calories,
  });

  //converts json
  factory NutritionProduct.fromJson(Map<String, dynamic> json) {
    return NutritionProduct(
      id: json['_id'] ?? '',
      name: json['name'] ?? '',
      price: (json['price'] as num).toDouble(),
      category: json['category'] ?? '',
      desc: json['desc'] ?? '',
      imageURL: json['imageURL'] ?? '',
      weightKg: (json['weightKg'] ?? 0).toDouble(),
      calories: (json['calories'] ?? 0).toDouble(),
    );
  }
}
